src/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.tsx

```diff
@@ -1,52 +1,84 @@
-import { FC, ReactNode } from 'react'
+import { FC, ReactNode, useCallback } from 'react'
 import { useTranslation } from 'react-i18next'
 import FilterListIcon from '@mui/icons-material/FilterList'
 import Typography from '@mui/material/Typography'
 import Button from '@mui/material/Button'
 import Box from '@mui/material/Box'
 
 import AppDrawer from '~/components/app-drawer/AppDrawer'
-import AppSelect from '~/components/app-select/AppSelect'
 import CheckboxList from '~/components/checkbox-list/CheckboxList'
 import FilterInput from '~/components/filter-input/FilterInput'
+import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
 
 import { spliceSx } from '~/utils/helper-functions'
+import { categoryService } from '~/services/category-service'
+import { subjectService } from '~/services/subject-service'
 import { styles } from '~/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.styles'
 import {
   ButtonVariantEnum,
+  CategoryNameInterface,
   CourseFilters,
+  CourseFiltersActions,
   PositionEnum,
   ProficiencyLevelEnum,
-  SizeEnum
+  SizeEnum,
+  SubjectNameInterface
 } from '~/types'
 
 interface CoursesFiltersDrawerProps {
+  additionalParams: Record<string, number | string | undefined>
+  filterActions: CourseFiltersActions<CourseFilters>
   filters: CourseFilters
-  handleFilterChange: (
-    key: keyof CourseFilters,
-    value: string | ProficiencyLevelEnum[]
-  ) => void
   onClose: () => void
   isOpen: boolean
-  handleReset: () => void
   deviceFields?: ReactNode
 }
-const fields = [
-  { value: 'music', title: 'Music' },
-  { value: 'lang', title: 'Languages' },
-  { value: 'development', title: 'Development' }
-]
 
 const CoursesFiltersDrawer: FC<CoursesFiltersDrawerProps> = ({
+  additionalParams,
+  filterActions,
   filters,
-  handleFilterChange,
   onClose,
   isOpen,
-  handleReset,
   deviceFields
 }) => {
   const levelOptions = Object.values(ProficiencyLevelEnum)
   const { t } = useTranslation()
+  const { updateFiltersInQuery, resetFilters } = filterActions
+
+  const getSubjectsNames = useCallback(
+    () => subjectService.getSubjectsNames(filters.category),
+    [filters.category]
+  )
+
+  const onCategoryChange = (
+    _: React.SyntheticEvent,
```

Comment:
if you use destructuring approach at file.
Can also destructure `SyntheticEvent`

Reply:
Fixed.